lib/spack/spack/build_environment.py

```diff
@@ -650,6 +650,9 @@ def setup_package(pkg, dirty):
         dpkg.setup_dependent_package(pkg.module, spec)
         dpkg.setup_dependent_environment(spack_env, run_env, spec)
 
+    parent_modules = parent_class_modules(pkg.__class__)
+    for mod in parent_modules:
+        set_module_variables_for_package(pkg, mod)
```

Comment:
653-656 and 643-646 should be factored out into a function or integrated into `set_module_variables_for_package`.